### PR TITLE
Price table

### PR DESCRIPTION
**Pull request by schneiderman805**
# PT Link

[set up price table ](https://www.pivotaltracker.com/story/show/163530895)

# Description of chore

adds pricing table to be able to choose price point and creates associations between sessions and price table.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -24,4 +31,16 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "users", force: :cascade do |t|
```
Comment (GergKllai1) [inline db/schema.rb:36]: This was dragged from another branch it is not part of this PR.
rails db:drop + rails db:create + delete schema file + rails:db migrate + rails db:seed

File: db/migrate/20190130183841_create_price_tables.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreatePriceTables < ActiveRecord::Migration[5.2]
+  def change
+    create_table :price_tables do |t|
+      t.integer :trainees
+      t.integer :price_point
+
```
Comment (GergKllai1) [inline db/migrate/20190130183841_create_price_tables.rb:6]: You set up the association in the DB but didn't set up the foreign key 
Sessions should have a foreign key price_table_id
You should create a migration for that

File: spec/models/price_table_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe PriceTable, type: :model do
+  describe 'DB table' do
+    it { is_expected.to have_db_column :trainees }
+    it { is_expected.to have_db_column :price_point }
+  end
+
```
Comment (GergKllai1) [inline spec/models/price_table_spec.rb:8]: You don't test the associations

File: spec/models/price_table_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :trainees }
+    it { is_expected.to validate_presence_of :price_point }
+  end
+
+  describe FactoryBot do
+    it 'should have valid Factory' do
+      expect(create(:price_table)).to be_valid
```
Comment (GergKllai1) [inline spec/models/price_table_spec.rb:20]: This test fails in semaphore

File: app/models/price_table.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class PriceTable < ApplicationRecord
+    validates_presence_of :trainees, :price_point
+    has_many :sessions
+    enum price: { low: 0, medium: 1, high: 2 }
```
Comment (GergKllai1) [inline app/models/price_table.rb]: This should be the same name as in the db table: `price_point`

**Review by GergKllai1 — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 1 above)
Comment (johannaanden) [reply db/schema.rb:36]: Done

File: db/migrate/20190130183841_create_price_tables.rb (hunk 2 above)
Comment (johannaanden) [reply db/migrate/20190130183841_create_price_tables.rb:6]: Done

File: app/models/price_table.rb (hunk 5 above)
Comment (johannaanden) [reply app/models/price_table.rb]: Done

File: spec/models/price_table_spec.rb (hunk 3 above)
Comment (johannaanden) [reply spec/models/price_table_spec.rb:8]: Done

File: spec/models/price_table_spec.rb (hunk 4 above)
Comment (johannaanden) [reply spec/models/price_table_spec.rb:20]: Done

**Review by kianaditya — APPROVED** [review]
LGTM!

**Review by GergKllai1 — APPROVED** [review]
LGTM!
